### ISSUE TITLE
AdoptOpenJDK - Download hangs on windows on 4/30/2019 (Edge & Chrome verified, multiple machines)

### ISSUE THREAD

**Issue by tcarlisle2012** [AdoptOpenJDK/openjdk-website#477]
There is something about your binaries downloads that hangs Windows antimalware  service executable. The downloads finish transferring data, but then the browser kicks it over to antimalware service executable to scan, which consumes CPU or GPU indefinitely and never returns. Once this happens, user cannot download files from other sites either because this keeps the antimlaware service executable hung. 

No other downloads from other sites do this. Verified on multiple PCs and multiple browsers.

**Comment by karianna** [comment]
@tcarlisle2012 Can you tell us which version of Windows you are using and the exact binary that you were trying to download?

**Comment by tcarlisle2012** [comment]
Sorry I put this in the wrong section. Both systems are running version 1809. With different build numbers. One is build 17763.348 and the other is 17763.427. 

I was originally trying to download OpenJDK11U-jdk_x64_windows_hotspot_11.0.3_7.msi, but then tried the .zip, then tried the 1.8LTS, and all do the same.

**Comment by adoptopenjdk-admin** [comment]
Andreas Ahlenstorf:   OpenJDK11U-jdk_x64_windows_hotspot_11.0.3_7.msi with Edge on a yesterday installed Windows 10 1809 with all patches downloaded without issues. It took the malware scanner some time, but it finished.
[1m]

Andreas Ahlenstorf:   The JDK 8 MSI just finished, too.

**Comment by karianna** [comment]
> Sorry I put this in the wrong section. Both systems are running version 1809. With different build numbers. One is build 17763.348 and the other is 17763.427.
> 
> I was originally trying to download OpenJDK11U-jdk_x64_windows_hotspot_11.0.3_7.msi, but then tried the .zip, then tried the 1.8LTS, and all do the same.

Have you perhaps got the Windows Defender settings higher than the default?

**Comment by tcarlisle2012** [comment]
No, Windows defender settings were set to the usual and unmodified. Of course, when I disable real-time protection, it then works, but that defeats the purpose. It doesnt matter if cloud based or auto sample submission options are o or off, it hangs as long as real time protection is enabled, and works when that is disabled. 

Another option, is to install a 3rd party like Avira, and then that product takes the front seat for real time protection. I installed Avira and verified no issue. 

But for people running windows defender as their primary real time protection might report issues with this site. Never had any other sites do something like this. 

Of course, if it scanned and failed that would give us something to work with, but it just hangs. Normally, when a download completes it gives a spinning indication that scanning is happening. That doesn't happen at all in the broken state. It just never even begins, as best I can tell.

**Comment by keithc-ca** [comment]
I've seen long running times for the scan, but it has always (eventually) finished. It may take minutes, but it does finish.

**Comment by karianna** [comment]
I'm unable to replicate a hang on my Windows 10 (via Virtual Box) either - Install was fine and it checked the London Jamocha Community Certificate.

@tcarlisle2012 Are you able to grab any Windows install / MSI / system logs?

**Comment by tcarlisle2012** [comment]
Ok, I think I've narrowed it. 

But first, is everyone chiming in using nothing but windows defender? Because if not then it really isn't comparing apples to apples. 

Using my test system, which only has windows defender, i moved the file to another web server and same thing. So we can rule out the web site or code. 

Looking further, if I right click the file (happens to be the zip file of your latest jdk11 x64 w/hotspot build) and choose to scan with Windows Defender, it scans indefinately. However, since I can see its status via the security settings front end, it is counting total files up in the 60,000 range now, and it is going to just keep going. Something about the archive is causing it to loop indefinitely in a scan by Windows Defender. 

Scanning with Avira on another system it finishes and counts the right number of files (604) and folders (131). 

Ok, while writing this, the windows defender scan of the extracted files did finish, counting a total of over 77,000 files. That can't be right, because there are not that many files extracted. Something is sending it into circles. 

Looking further, individually scanning each directory under the extratced jdk1.11.03 directory, they all complete in reasonable amounts of time, except the jmod directory takes about 15 minutes and counts thousands of files, yet there are only 71 files in that directory. 

So the bottom line is windows defender scanning is going in circles scanning the jmods directory. I have a feeling avira skips it altogether. Windows defender is probably providing better security posture, as scanning jmods is probably the right thing to do. 

But it should not scan in circles. Or perhaps it should.... I am not the expert in java > 1.8 and perhaps jmods call other jmods? If so, following that recursion is the right thing to do, but perhaps windows defender should track ones it already scanned. 

In any event, it is looking like a windows defender + jmod scanning issue and not a web site issue.

I did kick off a download of that built zip file (from the system running only windows defender) and am going to let it run until (or if) it finishes -- hours if that is what it takes. I will report the results here when that happens.  

On a side note, it looks like the files end up in the "downloads" directory as soon as they are done downloading, and before windows defender finishes its thing. But chrome & edge don't recognize it as complete until windows defender finishes. I mention this becuase people that are being held up with productivity because of this, since you are fairly sure the files are not malicious (you did check sha256 values, right?) you can continue working.

**Comment by karianna** [comment]
I was just using Windows Defender yes.  If it's scanning each class file I could see why it gets into thousands of files.  I'm pinging my contact at MSFT to see if they can help here.

**Comment by keithc-ca** [comment]
If defender is recursively scanning jmod files (which are very much like zip files), then the count should be higher, but I'd expect something closer to double the number of files, not 100 times as many.

**Comment by keithc-ca** [comment]
And if it scans the 'modules' file the count would be much higher (it contains >27,000 resources in a Linux x64 build).

**Comment by tcarlisle2012** [comment]
Good news, the system still running only windows defender did ultimately finish scanning the file after about 30 minutes. That system is way underpowered for the workload, but is the only one I have left without a third party security suite. But still, one would expect a downloaded msi or zip file to not take more than a few seconds. 

Bottom line, windows defender takes a lot longer because the archive contains jmods, and windows defender doesn't return success and control back to the browser until that is done, versus avira does and continues scanning in the background (and skips the jmods probably by default). 

I just saw the same downloading a build from jdk.java.net, so I am more than comfortable closing this and considering it an issue of "windows defender takes a very long time to scan JDK's"

**Comment by karianna** [comment]
Great thank you.

**Comment by douph1** [comment]
I confirm I have already seen same behaviour ( long long scan by windows defender ) for Oracle JDK.
I think Windows Defender is running/executing jvm code ( class ) into sandboxed env to detectect malicious activity

**Comment by karianna** [comment]
Going to reopen this and get MSFT folks to comment

**Comment by tresf** [comment]
Just ran into this bug... Interestingly enough, I have a project about 1/3 the size with quite a few class files as well as the entire Java FX runtime bundled and it takes seconds.

In my case, the project that does NOT suffer this issue is packaged as an EXE installer (not a MSI installer like AdoptOpenJDK).  Not sure if that information helps.  I can't imagine Microsoft would be OK with its OS spending 30 minutes scanning -- for example -- the .NET framework... this seems like something they'd be willing to help fix, assuming there's a proper escalation path.

![image](https://user-images.githubusercontent.com/6345473/57749723-4759d480-76ad-11e9-8aa3-bec3364131a4.png)

**Comment by tcarlisle2012** [comment]
Just to provide info, I was downloading the zipped JDK, not the installer version. Also, I have downloaded JDK from Oracles site and got the same issue. So I am fairly confident this is an issue with Windows Defender and how it handles java .MOD files. 

I would venture a guess that third party AV solutions, like Avira that I used to use, obtain better performance by not scanning java classes recursively. Clearly that would result in a lower security posture. Or, cloud based security features handle it by only scanning the file from the download source the first time the cloud sees that file signature, and upon determining it clean after that point the AV considers it trusted and other users are spared the scan. That would be better than skipping java archives altogether, but still be less secure than 100% scanning. 

I have gotten into the practice if turning Defender off when downloading a java archive from a reputable source.

**Comment by tresf** [comment]
> > > assuming there's a proper escalation path.

> > Going to reopen this and get MSFT folks to comment

> this is an issue with Windows Defender and how it handles java .MOD files.

@karianna says MSFT is a sponsor to the AdoptOpenJDK project, so they're best to help with this.  This may be anecdotal, but Firefox didn't seem to trigger the 30-minute download/scan/delay issue, so that may be a stop-gap for users in addition to modifying the Windows Defender scan settings.

**Comment by jesuiscamille** [comment]
I got the same problem with the "OpenJDK11U-jdk_x64_windows_hotspot_11.0.3_7.msi" file.
Well, I think... The download process hung at the end of the file-fetch. (I didn't verify if it was due to some antivirus scan).

I'm trying to fetch it with Firefox to see if I get another result (as tresf stated)

**Comment by tcarlisle2012** [comment]
Any decent browser is going to toss the file to Windows AV, which will toss
it to a 3rd party AV if installed -- and the same issue will take place.
You will notice the same behavior if you go to oracle's site and download a
zipped JDK from them, and most likely any java archive. I've searched
google for issues similar and people report this behavior with many
different types of zipped files -- like MalwareBtyes application.

There really is only one what to speed it up, which would be to have it not
dive into each file and each call to other files recursively... or to trust
the file completely and not scan it or scan it lightly to make sure its
contents do match up with a signature of a version that has been deemed
safe. One would think that the cloud based features in Windows Defender
would be doing exactly this, but apparently not. The 3rd party solutions
probably are doing a much, much better job at this.

Since they don't provide a way of turning off this recursion, your only
options (if you don't want to move to a 3rd party solution) are to exclude
a directory or a process or turn off real-time altogether (temporarily).
The problem with exclusions is that if you exclude your downloads directory
or the browser process, that is really the main point of ingress where
nasty code will come in from the external world. I am more comfortable
recommending to turn off real-time protection when downloading a file that
is problematic this way which you are comfortable trusting.

Or just wait. I suppose that is an option. It takes about 20 minutes on up
depending on the system. MS Defender is not designed to take all compute
resource available in order to expedite, because if it did the performance
footprint would be way to high and no one would use it. It seems to max out
at about 25% of my CPU on a modern Core-I7 and it never pushes disk IO
anywhere near 100%.  Hence by design it is throttled and hence these scans
take a long time.

**Comment by tresf** [comment]
> which will toss it to a 3rd party AV if installed -- and the same issue will take place.

Tested with McAfee -- for some reason when downloading with Chrome or Edge the default behavior is to NOT iterate through `Data1.cab` and scan each internal file.  Even when scanned manually, the scan takes about 3 minutes (NOT 30 minutes).

> > Firefox didn't seem to trigger the 30-minute download/scan/delay issue, so that may be a stop-gap for users in addition to modifying the Windows Defender scan settings.
> Any decent browser is going to toss the file to Windows AV

Just reiterating that Firefox doesn't. :D

**Comment by tcarlisle2012** [comment]
You are correct, Firefox doesn't scan downloads. Just tested with the
proverbial eicar test file, and firefox will allow you to download it
without any issues. It used to be common for firefox users to experience
download delays and the solution was to disable scanning. I suppose that
since firefox is putting speed as its highest priority these days, that
they err on the side of speed in this regard.

**Comment by brainplot** [comment]
I'm experiencing the same behavior. It's not taking 30 minutes in my case though. I downloaded `OpenJDK8U-jdk_x64_windows_hotspot_8u212b03.msi` 2 hours ago and Chrome doesn't yet report the download as completed.
I have a desktop machine with an i5-8600k and 16 GB of RAM, which should be plenty of computing power. Not sure if I should just wait or force-quit the process and retry with real-time protection disabled.

**Comment by karianna** [comment]
> I'm experiencing the same behavior. It's not taking 30 minutes in my case though. I downloaded `OpenJDK8U-jdk_x64_windows_hotspot_8u212b03.msi` 2 hours ago and Chrome doesn't yet report the download as completed.
> I have a desktop machine with an i5-8600k and 16 GB of RAM, which should be plenty of computing power. Not sure if I should just wait or force-quit the process and retry with real-time protection disabled.
> 
> **EDIT**: Windows 10 1809 - 17763.529

I'd try through Firefox as a workaround and if that fails then disable protection for that one download.

**Comment by brainplot** [comment]
> if that fails then disable protection for that one download

That's what I ended up doing, eventually. I left it running for 3 and a half hours but it wouldn't finish scanning. However, I would like to point out that once I got the file downloaded, without real-time protection, I manually scanned the file with Windows Defender and it took 2-3 minutes. The scan reported something like 70k files that were scanned. I think the real-time protection process is assigned such a low priority and I/O bandwidth that it cannot scan that many files in a reasonable amount of time.

**Comment by frevds** [comment]
Same here for AdoptOpenJDK8, Intel Xeon 3GHz and 32GB, all newest Windows updates. 
Windows Defender keeps scanning indefinitely with 600mb of RAM usage and full use of all cores and disk. 
I entered the properties of the downloaded file and unblocked it (to be able to start it even though it came from an internet location). 
Looks like this is a bug in Windows Defender rather than the JDK binaries (after all it hangs).

**Comment by SirTwsted** [comment]
I can confirm this on July 6th 2019 6:10-am EDT.
This does not want to download, it stays stuck at the very very end and hangs there forever.
It does not want to finish & it somewhat makes it impossible to surf, because it half locks the browser.
As the browser is waiting for the download task to end,  but it doesnt.
Forcing me to Force hard boot the computer, because its the only way to break from it.
I tried ending the task in task manager, but when I do, that then hangs the task manager.
This is just a pure headache right now.

**Comment by htfv** [comment]
I can confirm the issue, but I also want to add one detail. After Chrome finishes downloading, it hangs there forever. But at the same time, I can go to download folder, extract all files, and start using JDK without any issues.

I don't know exactly how Chrome+Defender work together, but it seems like Defender has no problem when JDK files are extracted from zip. To confirm that real-time protection is active, I created the EICAR test file - Defender reacted immediately and suggested to remove the file.

This makes me think that Defender has problem with the zip file itself, not the files contained in the zip.

**Comment by frevds** [comment]
Yes, it's a bug in Defender and should be brought to @Microsoft's attention. It might be security relevant, if Defender is not reacting on other threats any more (but you @htfv said it would be watching even if it hangs with a download). At least it's  not benefitial for system stability, as it prevents new downloads and stalls the CPU. So everybody can at least crash the system with such a ZIP file, if not crash Defender.

**Comment by karianna** [comment]
Hi all - This has been passed onto the Windows Defender team - hope to hear back in a few weeks.

**Comment by tresf** [comment]
To identify scope, I decided to run a test on other 64-bit installers.  @karianna feel free to share this information with Microsoft if you think it will add value.  As a stop-gap, it may be beneficial to do a comparison on the better benchmarking installers (hint it's not file size as BellSoft and Amazon are 80MB larger than Adopt's) 

#### Environment

Tests were done on a brand new HP Spectre i7 Quad Core (8 logical processors) Laptop running Windows 10 Professional 64-bit from Microsoft (NOT the verison HP ships).
* All tests had similar symptoms: The CPU was ~12% CPU using Task Manager "Details" Tab, ~130MB RAM.  (100% / 8 logical processors = 12.5% per logical processor).   Note, the "Process" tab shows a 
 lower value, 8% - 10% CPU.
* The laptop was running from battery 🔋, so benchmarks may be influenced slightly by CPU throttling.
* In all tests, I confirmed `MsMpEng.exe` was the top CPU process (no other high-CPU tasks stealing priority).

#### Results
In order of severity (as and end-user I would only consider < 5 minutes to be "acceptable" results)

* 🐞 AdoptOpenJDK JDK 11 MSI (Gave up after 3 hours)
* 🐞 SapMachine JDK 11 ZIP (Gave up waiting after 35 minutes)
* ⚠️ RedHat JDK 11 MSI (Finished scanning in under **15** minutes)
* ⚠️ Azul Zulu JDK 11 MSI (Finished scanning in under **10** minutes)
* ⚠️ Amazon Correto JDK 11 MSI (Finished scanning in under **10** minutes)
* ⚠️ BellSoft Liberica JDK 11 MSI (Finished scanning in under **10** minutes)
* ✅ Oracle JDK 11 EXE (Finished scanning in under **5** minutes)

**Comment by karianna** [comment]
That's some amazing analysis @tresf - thank you.  Hmmmm, that tells me that our MSI's are either shaped differently or are signed differently.

@ali-ince Can you do a compare and contrast for us?

**Comment by karianna** [comment]
Hi all - had word back from the Defender team, who should have improved this already (please get your latest windows updates), with more improvements next month. Please let me know if you see improvement or not when you have time.

**Comment by tresf** [comment]
> who should have improved this already

Any version specificites to share (e.g.)?
* `KB2267602`: **`1.297.1192.0`** (July 16)
* `KB2267602`: **`1.297.1087.0`** (July 14)
* `KB2267602`: **`1.297.956.0`** (July 12)

I tested with the following URLs:
* https://github.com/AdoptOpenJDK/openjdk11-binaries/releases/download/jdk-11.0.3%2B7/OpenJDK11U-jdk_x64_windows_hotspot_11.0.3_7.msi<br><br>✅ **Download finishes immediately**<br><br>
* https://github.com/AdoptOpenJDK/openjdk11-binaries/releases/download/jdk11u-2019-07-16-04-03/OpenJDK11U-jdk_x64_windows_hotspot_2019-07-16-04-03.msi<br><br>🐞**Gave up after 10 minutes**<br><br>

@karianna Is it safe to assume they simply whitelisted the official installer this time around?  And is it safe to assume that by "more improvements next month", they're going to permanently fix the scanning engine to address this bug for all installers moving forward?

**Comment by karianna** [comment]
@tresf I wonder if in the release stage we sign our binary in a different way, else as you say they've whitelisted our releases.

I think they're looking at all installers yes.

**Comment by mjpt777** [comment]
I've just tried 8u222 and it is having the issue again.

**Comment by karianna** [comment]
Thanks, Martin, I'll ping them again - they may be whitelisting us as we put new releases out.

**Comment by mjpt777** [comment]
The latest is running for a few hours now at 30+ MBs to disk and eating a whole CPU. :-(

**Comment by gdams** [comment]
my understanding as that as long as the binary is signed with the same cert it should already be whitelisted. @karianna can you check whether they whitelisted our cert or a single binary?

**Comment by gdams** [comment]
@mjpt777 did you see any defender warnings when you went to open the installer? I'd be keen to know whether you see the warning below or not? Supposedly according to the Microsoft docs you can only remove this if you use an EV codesign cert which we can't as it requires a randomly changing token to unlock it. The docs do say that with a standard codesign cert you will stop seeing that warning once the certificate has been downloaded/trusted on a certain number of systems.

![](https://filestore.community.support.microsoft.com/api/images/c3ae183d-b756-4432-8c7b-ed12415f7b24?upload=true)

**Comment by gdams** [comment]
@mjpt777 I created a windows 10 VM in azure and had no issue download and running the following installer using edge: https://github.com/AdoptOpenJDK/openjdk11-binaries/releases/download/jdk11u-2019-07-16-04-03/OpenJDK11U-jdk_x64_windows_hotspot_2019-07-16-04-03.msi

Must have taken seconds from clicking downkload and run to the actual installer opening

**Comment by tresf** [comment]
@gdams just for clarification, the message above from @mjpt777 says 8u222.  It was my post before that which reported 11u2019-07-16-04.  I'll retest both now.

Regarding EV: Yes, EV certs are a manual process unless you use a dedicated appliance and have someone sign into it periodically.  It may be worth it for official releases.

Note, Apple's notarization (in addition to the signing requirements required today) and stapling (essentially, timestamping) adds even more complexity (although with the right tools can be automated).  It's off-topic to this thread, but it's going to be required in Catalina.  https://developer.apple.com/documentation/security/notarizing_your_app_before_distribution.

**Comment by tresf** [comment]
Defender has provided the following updates since my last testing:

* `KB2267602`: **`1.297.1248.0`** (July 17)
* `KB2267602`: **`1.297.1193.0`** (July 16)


* ❓@mjpt777 I was unable to locate 8u222, can you please provide a direct link for testing?
* 🐞@gdams I've updated to the above Windows Defender patches and the comments I've posted in https://github.com/AdoptOpenJDK/openjdk-website/issues/477#issuecomment-511848658 still remain true.  I have not yet waited the full 10 minutes, but your testing results do not match my testing results.

**Comment by tresf** [comment]
Sorry, Defender platform found another update (unsure why this wasn't found the previous iteration).  Running tests again.
   * `KB052623`: **`4.18.1907.4`** (July 17) (Platform Update -- NOT a definition update)

**Comment by karianna** [comment]
Oh, I think I might have an idea - @mjpt777 was probably using the jdk8u222-b10 upstream build that Red Hat produces but distributes through Adopt.  I suspect that does not have a signing cert.

The Adopt 8u222-b10 binary is still coming out of our pipelines.

**Comment by tresf** [comment]
Test results:
* 🐞The latest Windows Defender Platform Update `4.18.1907.4` (NOT a definition update) update does **NOT** address the issue with the [nightly installer](https://github.com/AdoptOpenJDK/openjdk-website/issues/477#issuecomment-512355952).  It's been scanning for 5 minutes now... 

@gdams I'm not sure how Azure handes Windows defender scans.  Was this platform suitable for reproducing previous issues?  If so, comparing defender definitions and scanning engines may be warranted if you cannot reproduce.

**Comment by mjpt777** [comment]
The link I'm using is: https://github.com/AdoptOpenJDK/openjdk8-upstream-binaries/releases/download/jdk8u222-b10/OpenJDK8u-jdk_x64_windows_8u222b10.zip

It has been running for a few hours now and my laptop is over heating! I did not get the "Windows Protected up PC" screen.

**Comment by mjpt777** [comment]
BTW a similar issue has been observed with the Zulu JDK.

**Comment by karianna** [comment]
> BTW a similar issue has been observed with the Zulu JDK.

And they sign theirs I'd imagine but we should confirm that.

within 24 hours we should get proper signed adopt binaries out and we can see how they fare (see if they're whitelisted)

**Comment by gdams** [comment]
> And they sign theirs I'd imagine but we should confirm that.

They do indeed.

> within 24 hours we should get proper signed adopt binaries out and we can see how they fare (see if they're whitelisted)

There is no difference between a release and nightly build in terms of codesigning.

**Comment by karianna** [comment]
Hmm, in that case, I can only imagine whitelisting and/or a difference in the windows defender version on people's machines (although I'd expect most techies to aggressively upgrade that component).

**Comment by douph1** [comment]
Windows 10 + Chrome and definition update : KB2267602: 1.297.1299.0 from today and analysis *still running* after 1 hour for OpenJDK12U-jdk_x86-32_windows_hotspot_12.0.1_12.msi
It is worse than before .. usually it was between 3min(yesterday) and 20 minutes.

It can't be a signer whitlist ... and as people complain about scanning time for zip file ( wich is not signed ) event whitelisting our cert will not be sufficient

**Comment by wormuth** [comment]
Thought me and my teammates where the only people facing this problem...

Most downloads of significant size (>100m) via Chrome are blocking for us. No way out, yet but using FF for those kind of download :-/

So I would say this issue is definitely not specific to your AdoptOpenJDK archive.

**Comment by mjpt777** [comment]
The latest signed build downloads and installs without issue. However when I attempt to delete the installer file (Shift + Del) the problem resurfaces with it taking forever to calculate how long to delete. I have to reboot to make progress.

**Comment by sazsaz** [comment]
Having the same issue with Microsoft Edge 44.17763.1.0 / Chrome 75.0.3770.142. The JDK file `https://github.com/AdoptOpenJDK/openjdk8-binaries/releases/download/jdk8u222-b10_openj9-0.15.1/OpenJDK8U-jdk_x64_windows_openj9_8u222b10_openj9-0.15.1.zip`  hangs while scanning using Windows Defender.

Incidentally there is no issue when using Firefox.

**Comment by karianna** [comment]
Hi all,

FYI - Windows defender is due another update in the next 24-48 hours and then a larger fix will occur in August.

**Comment by stepheneb** [comment]
Seeing same issue downloading OpenJDK on Microsoft Edge 44.18362.1.0. After downloading scanning continued for over an hour before I terminated the operation.

Downloaded and saved fine with Firefox and ran with Windows Installer. It took about 10 minutes before I got the dialog that OpenJDK was asking permission to install itself and then another 5m to complete.

**Comment by aahlenst** [comment]
@karianna I just tested freshly generated Windows 10 1903 images. All results with Edge, no third-party AV:

JDK 8u222b10 MSI
JDK 11.0.4+11 MSI 
JDK 11.0.4+11 ZIP
JDK 12.0.2+10_openj9-0.15.1 ZIP

all complete within 4 to 5 minutes on a machine with 4 CPUs and 16 GB RAM.

**Comment by karianna** [comment]
> @karianna I just tested freshly generated Windows 10 1903 images. All results with Edge, no third-party AV:
> 
> JDK 8u222b10 MSI
> JDK 11.0.4+11 MSI
> JDK 11.0.4+11 ZIP
> JDK 12.0.2+10_openj9-0.15.1 ZIP
> 
> all complete within 4 to 5 minutes on a machine with 4 CPUs and 16 GB RAM.

Thanks for confirming!  Can others check (please make sure your O/S, Windows defender, and browser are at the latest versions.

**Comment by badgerwithagun** [comment]
Crazy-long scan now on two different machines, but it _does_ eventually complete.  I've not encountered an indefinite hang yet (although I did nearly give up on the second one).  Definitely the longest scan I've seen on pretty much anything.

It seems that the browser has a single thread dedicated to download management, because it blocks that and stops anything else being downloaded in the meantime, but that's a bug for someone else!

Windows Version 10.0.18362 Build 18362
MSMpEng 4.18.1907.4-0
Brave Browser Version 0.68.131 Chromium: 76.0.3809.100 (Official Build) (64-bit)

**Comment by karianna** [comment]
I'm now working for Microsoft :-).  I'm going to find the right team and track this down.

**Comment by karianna** [comment]
I've found the correct folks.  The update still needs to be rolled out to 100% of windows users.  I'll check back here in a week

**Comment by tresf** [comment]
FYI, Defender is coming to Linux, so the problem may cascade to other platforms if the factors which influence this are common between AV engines/archives https://www.zdnet.com/article/microsoft-defender-atp-is-coming-to-linux-in-2020/

**Comment by karianna** [comment]
@tcarlisle2012 Have you had a chance to try again recently?

**Comment by tresf** [comment]
> @tcarlisle2012 Have you had a chance to try again recently?

I've been working with AdopOpenJDK on Windows right along and I had completely forgotten this bug, suggesting it has been fixed.  Today specifically, I installed 11.0.6, 8.0.242 (both JDK and JRE) and no endless wait times.

I have a feeling that the bug will rear its ugly head again someday in the future, but the MSIs don't seem to be having issues any more. 🤞

**Comment by aahlenst** [comment]
@tresf Thanks for confirming.

If the problem comes back, please do not hesitate to reopen.

**Comment by gdams** [comment]
Issues have been reported with the new codesign cert. Sounds like it needs to be whitelisted by the defender team

**Comment by jmichler** [comment]
Hi,
I think this is still an issue:
C:\Program Files (x86)\Horus_269\jre18282\bin>java -version
openjdk version "1.8.0_282"
OpenJDK Runtime Environment (AdoptOpenJDK)(build 1.8.0_282-b08)
OpenJDK 64-Bit Server VM (AdoptOpenJDK)(build 25.282-b08, mixed mode)

C:\Program Files (x86)\Horus_269\jre18282\bin>C:\Users\jmichler\Downloads\Sigcheck\sigcheck64.exe -a java.exe

Sigcheck v2.80 - File version and signature viewer
Copyright (C) 2004-2020 Mark Russinovich
Sysinternals - www.sysinternals.com

C:\Program Files (x86)\Horus_269\jre18282\bin\java.exe:
        Verified:       Signed
        Signing date:   14:23 20.01.2021
        Publisher:      London Jamocha Community CIC
        Company:        AdoptOpenJDK
        Description:    OpenJDK Platform binary
        Product:        OpenJDK Platform 8
        Prod version:   8.0.2820.8
        File version:   8.0.2820.8
        MachineType:    64-bit
        Binary Version: 8.0.2820.8
        Original Name:  java.exe
        Internal Name:  java
        Copyright:      Copyright ® 2021
        Comments:       n/a
        Entropy:        6.526


I'm having this for an app based on eclipse RCP; maybe it is only then? See also https://answers.microsoft.com/en-us/windows/forum/windows_10-security/first-start-of-eclipse-rich-client-based/42ada574-acba-49e2-b1ad-0f585b747fa3?tm=1618590387660

**Comment by karianna** [comment]
please raise this at gihub.com/adoptium if it is still an issue.
